tagstudio/src/core/media_types.py

```diff
@@ -0,0 +1,495 @@
+        is_iana=False,
+    )
+    PDF_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.PDF,
+        extensions=_PDF_SET,
+        is_iana=False,
+    )
+    PLAINTEXT_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.PLAINTEXT,
+        extensions=_PLAINTEXT_SET,
+        is_iana=False,
+    )
+    PRESENTATION_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.PRESENTATION,
+        extensions=_PRESENTATION_SET,
+        is_iana=False,
+    )
+    PROGRAM_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.PROGRAM,
+        extensions=_PROGRAM_SET,
+        is_iana=False,
+    )
+    SHORTCUT_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.SHORTCUT,
+        extensions=_SHORTCUT_SET,
+        is_iana=False,
+    )
+    SOURCE_ENGINE_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.SOURCE_ENGINE,
+        extensions=_SOURCE_ENGINE_SET,
+        is_iana=False,
+    )
+    SPREADSHEET_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.SPREADSHEET,
+        extensions=_SPREADSHEET_SET,
+        is_iana=False,
+    )
+    TEXT_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.TEXT,
+        extensions=_DOCUMENT_SET | _PLAINTEXT_SET,
+        is_iana=True,
+    )
+    VIDEO_TYPES: MediaCategory = MediaCategory(
+        media_type=MediaType.VIDEO,
+        extensions=_VIDEO_SET,
+        is_iana=True,
+    )
+
+    ALL_CATEGORIES: list[MediaCategory] = [
+        ADOBE_PHOTOSHOP_TYPES,
+        AFFINITY_PHOTO_TYPES,
+        ARCHIVE_TYPES,
+        AUDIO_MIDI_TYPES,
+        AUDIO_TYPES,
+        BLENDER_TYPES,
+        DATABASE_TYPES,
+        DISK_IMAGE_TYPES,
+        DOCUMENT_TYPES,
+        FONT_TYPES,
+        IMAGE_ANIMATED_TYPES,
+        IMAGE_RAW_TYPES,
+        IMAGE_TYPES,
+        IMAGE_VECTOR_TYPES,
+        INSTALLER_TYPES,
+        MATERIAL_TYPES,
+        MODEL_TYPES,
+        PACKAGE_TYPES,
+        PDF_TYPES,
+        PLAINTEXT_TYPES,
+        PRESENTATION_TYPES,
+        PROGRAM_TYPES,
+        SHORTCUT_TYPES,
+        SOURCE_ENGINE_TYPES,
+        SPREADSHEET_TYPES,
+        TEXT_TYPES,
+        VIDEO_TYPES,
+    ]
+
+    @staticmethod
+    def get_types(ext: str, mime_fallback: bool = False) -> set[MediaType]:
+        """Returns a set of MediaTypes given a file extension.
+
+        Args:
+            ext (str): File extension with a leading "." and in all lowercase.
+            mime_fallback (bool): Flag to guess MIME type if no set matches are made.
+        """
+        types: set[MediaType] = set()
+        # mime_guess: bool = False
+
+        for cat in MediaCategories.ALL_CATEGORIES:
+            if ext in cat.extensions:
+                types.add(cat.media_type)
+            elif mime_fallback and cat.is_iana:
+                type: str = mimetypes.guess_type(Path("x" + ext), strict=False)[0]
+                if type and type.startswith(cat.media_type.value):
```

Comment:
`type` is a name of builtin, so it would be better to use different variable name.